src/app/app.component.ts

```diff
@@ -34,4 +49,84 @@ export class AppComponent {
   async accountManagement() {
     await this.authService.account();
   }
+
+  getAccount() {
+    this.apiService.getAccount().subscribe(res => {
+      this.userAccount = res;
+    });
+  }
+
+  openDialog(): void {
+    const dialogRef = this.dialog.open(CreateGroupDialog, {
+      width: '250px',
+      data: {name: this.name, description: this.description}
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
+      console.log('The dialog was closed', result);
+      this.name = result;
+    });
+  }
+
+  getUserGroups() {
+    this.getAccount();
+    this.apiService.getGroups().subscribe(res => {
+      this.apiService.getAccountGroups(this.userAccount.id).subscribe(res => {
+        this.userAccountGroups = res;
+        for (var group of this.userAccountGroups) {
+          console.log("group:");
+          console.log(group);
+          this.apiService.getGroup(group.GroupId).subscribe(res => {
+            this.userGroups.push(res);
+          },
+          err => {
+            console.log(err);
+          })
+        }
+      },
+      err => {
+        console.log('err', err);
+      });
+    });
+  }
+}
+
+@Component({
+  selector: 'create-group-dialog',
+  templateUrl: 'group.page.dialog.html',
+})
+export class CreateGroupDialog {
+
+  constructor(
+    public apiService: ApiService,
+    public dialogRef: MatDialogRef<CreateGroupDialog>,
+    @Inject(MAT_DIALOG_DATA) public data: DialogData
+  ) {  }
+
+  onNoClick(): void {
+    this.dialogRef.close();
+  }
+
+  createGroup(group): void {
+    this.apiService.createGroup(group).subscribe(
+      groupResponse => {
+        let groupId = groupResponse.body.id;
+      },
+    )
+  }
+
+  submit() {
+   let group: Group = {};
+
+   group['title'] = this.data.title;
+   group['short_description'] = this.data.short_description;
+   group['description'] = this.data.description;
+
+   this.apiService.getAccount().subscribe(
+     response => {
+       group['AdminId'] = response.id;
+       this.createGroup(group);
+     },
+   );
+ }
```

Comment:
This should be its own component in the /src/app/_components folder. Also, it feels a bit weird to have its logic in the app.component. I'm thinking it might be better to just make a blank page for it, have the button go to the page, and then have the ngOnInit of that page auto-open the dialog. Sorry it's kind of convoluted. We don't have any example of dialogs on core, and I've only ever triggered them from a page giving them context (not a sidebar/the main app component)